index.js

```diff
@@ -3656,6 +3663,50 @@
       [Fn(a, b), m(a), m(a)],
       Z.sortBy);
 
+  //# zipWith :: (a -> b -> c) -> Array a -> Array b -> Array c
+  //.
+  //. Returns the result of combining, pairwise, the given arrays using the
+  //. given binary function. The length of the resulting array is equal to the
+  //. length of the shorter input array.
+  //.
+  //. ```javascript
+  //. > S.zipWith(a => b => a + b, ['a', 'b'], ['x', 'y', 'z'])
+  //. ['ax', 'by']
+  //.
+  //. > S.zipWith(a => b => [a, b], [1, 3, 5], [2, 4])
+  //. [[1, 2], [3, 4]]
+  //. ```
+  function zipWith(f, xs, ys) {
+    var result = [];
+    var len = Math.min(xs.length, ys.length);
+    for (var idx = 0; idx < len; idx += 1) result.push(f(xs[idx])(ys[idx]));
+    return result;
+  }
+  S.zipWith =
+  def('zipWith',
+      {},
+      [Fn(a, Fn(b, c)), $.Array(a), $.Array(b), $.Array(c)],
+      zipWith);
+
+  //# zip :: Array a -> Array b -> Array (Pair a b)
+  //.
+  //. Returns the result of combining, pairwise, the given arrays using the
+  //. given binary function. The length of the resulting array is equal to the
+  //. length of the shorter input array.
+  //.
+  //. ```javascript
+  //. > S.zip(['a', 'b'], ['x', 'y', 'z'])
+  //. [['a', 'x'], ['b', 'y']]
+  //.
+  //. > S.zip([1, 3, 5], [2, 4])
+  //. [[1, 2], [3, 4]]
+  //. ```
+  S.zip =
+  def('zip',
+      {},
+      [$.Array(a), $.Array(b), $.Array($.Pair(a, b))],
+      S.zipWith(pair));
```

Comment:
Let's avoid using `S` functions internally. You'll need to define `zip` in terms of `zipWith`.